### PR TITLE
Integrate with resource dispatcher

### PR DESCRIPTION
**Pull request by misohu**
Kserve now can be integrated with resource dispatcher to be able to serve models from MLflow. In order to do that following manifests need to be created in user namespaces (secret and svc abbount with credentials to connect to object store) https://kserve.github.io/website/0.11/modelserving/storage/s3/s3/#deploy-the-model-on-s3-with-inferenceservice 

This PR:
- allows kserve to relate to minio (this is optional relation because kserve can work also without it).
- sends secrets and service accounts to resource dispatcher through relations

Integration tests are deploying metacontroller, poddefaults crd minio and resource dispatcher to simulate the usecase. An example labeled user namespace is created and the test check that the user manifests are created.

### CODE REVIEW

File: charms/kserve-controller/metadata.yaml [hunk 1]
```diff
@@ -24,9 +24,63 @@ resources:
...
+            service:
+              type: string
+          required:
+          - access-key
+          - port
+          - secret-key
+          - secure
+          - service
+    versions: [v1]
+    __schema_source: https://raw.githubusercontent.com/canonical/operator-schemas/master/object-storage.yaml
```
Comment (kimwnasptd) [inline charms/kserve-controller/metadata.yaml:55]: @misohu doesn't it mean that if we use `__schema_source` then we are not working in airgap?

Comment (misohu) [reply charms/kserve-controller/metadata.yaml:55]: Above  it is the whole schema spec which is used in airgaped.

File: charms/kserve-controller/src/charm.py [hunk 2]
```diff
@@ -60,6 +70,19 @@
     "src/templates/webhook_manifests.yaml.j2",
 ]
 
+# Values for MinIO manifests https://kserve.github.io/website/0.11/modelserving/storage/s3/s3/
+S3_USEANONCREDENTIALS = "false"
+S3_REGION = "us-east-1"
+S3_USEHTTPS = "0"
```
Comment (i-chvets) [inline charms/kserve-controller/src/charm.py:76]: Are we ok with hardcoding these?

**Comment by i-chvets** [comment]
Minor comment. LGTM otherwise.

File: charms/kserve-controller/src/charm.py (hunk 2 above)
Comment (misohu) [reply charms/kserve-controller/src/charm.py:76]: Yes for now. Opened [issue](https://github.com/canonical/kserve-operators/issues/186) to explain

**Comment by misohu** [comment]
One minor fix in credential. Plus I created [issue](https://github.com/canonical/kserve-operators/issues/186) to document hardcoded values for secret.

**Review by i-chvets — APPROVED** [review]
Approved.

**Pull request merged**